### PR TITLE
SCOAP3 affiliations export tool fix;

### PR DESCRIPTION
**Pull request by ErnestaP**
* affiliations export tool was inconsistent: returning different results every other time
* problem solved by using the scroll for searching through ES pages
* ref: https://github.com/cern-sis/issues-scoap3/issues/112
* data retrieved by fixed affiliations export, country="Sweden", year="2022": https://docs.google.com/spreadsheets/d/1JRHaa0o0Bc02qvNaOSgXFbX6MR7BzyVVk9ftgI1e5OI/edit#gid=1418757323

### CODE REVIEW

File: scoap3/modules/tools/tools.py [hunk 1]
```diff
@@ -39,74 +37,69 @@ def affiliations_export(country=None, year=None):
...
 
     # query ElasticSearch for result (and get total hits)
     query = get_query_string(country=country, year=year)
     search_results = current_search_client.search(
-        q=query, index=search_index, _source=source_fields, size=size, from_=index
-    )
+        q=query, index=search_index, _source=source_fields, scroll='1m')
+
+    sid = search_results['_scroll_id']
+    scroll_size = len(search_results['hits']['hits'])
```
Comment (drjova) [inline scoap3/modules/tools/tools.py:47]:
```suggestion
    scroll_size = results['hits']['total']['value']
```

File: scoap3/modules/tools/tools.py [hunk 2]
```diff
@@ -39,74 +37,69 @@ def affiliations_export(country=None, year=None):
...
         year if year else 'ALL'
     ))
     logger.info('Total results from query: {}'.format(total_hits))
 
-    if total_hits == 0:
+    if scroll_size == 0:
         return {'header': result_headers, 'data': result_data}
 
-    while index < total_hits:
+    while (scroll_size > 0):
```
Comment (drjova) [inline scoap3/modules/tools/tools.py]:
```suggestion
    while scroll_size > 0:
```

File: scoap3/modules/tools/tools.py [hunk 3]
```diff
@@ -39,74 +37,69 @@ def affiliations_export(country=None, year=None):
...
+        for record_index in range(scroll_size):
```
Comment (drjova) [inline scoap3/modules/tools/tools.py]: you just need  iterate through the results:

```
for result in search_results['hits']['hits']:
 ....
 
 ```
 
 
 this way you don't need to keep an index

File: scoap3/modules/tools/tools.py [hunk 4]
```diff
@@ -39,74 +37,69 @@ def affiliations_export(country=None, year=None):
...
                       'country', 'affiliation', 'authors with affiliation', 'total number of authors']
     result_data = []
-    index = 0
 
     # query ElasticSearch for result (and get total hits)
     query = get_query_string(country=country, year=year)
     search_results = current_search_client.search(
-        q=query, index=search_index, _source=source_fields, size=size, from_=index
-    )
+        q=query, index=search_index, _source=source_fields, scroll='1m')
```
Comment (drjova) [inline scoap3/modules/tools/tools.py]: why the size is missing? it will get the default from ES which I believe it's `10`

Comment (drjova) [reply scoap3/modules/tools/tools.py]: Also I would suggest to keep the scroll `5m`

File: scoap3/modules/tools/tools.py [hunk 5]
```diff
@@ -39,74 +37,69 @@ def affiliations_export(country=None, year=None):
...
-                result_data.append(
-                    [year, journal, doi, arxiv, arxiv_category, aff_country, aff_value, count, total_authors]
-                )
+                # add extracted information to result list
+                for meta, count in extracted_affiliations.items():
+                    aff_value, aff_country = meta
+                    result_data.append(
+                        [year, journal, doi, arxiv, arxiv_category, aff_country, aff_value, count, total_authors]
+                    )
+        search_results = current_search_client.scroll(scroll_id=sid, scroll='2m')
```
Comment (drjova) [inline scoap3/modules/tools/tools.py]:
```suggestion
        search_results = current_search_client.scroll(scroll_id=sid, scroll='5m')
```

File: scoap3/modules/tools/tools.py (hunk 1 above)
Comment (ErnestaP) [reply scoap3/modules/tools/tools.py:47]: But the scroll size has to be the same size as the page size or smaller

Comment (ErnestaP) [reply scoap3/modules/tools/tools.py:47]: Screenshot 2023-01-26 at 11 28 02

Comment (drjova) [reply scoap3/modules/tools/tools.py:47]: true!

File: scoap3/modules/tools/tools.py [hunk 6]
```diff
@@ -29,84 +29,76 @@ def affiliations_export(country=None, year=None):
...
-                )
+                # add extracted information to result list
+                for meta, count in extracted_affiliations.items():
+                    aff_value, aff_country = meta
+                    result_data.append(
+                        [year, journal, doi, arxiv, arxiv_category, aff_country, aff_value, count, total_authors]
+                    )
+        search_results = current_search_client.scroll(scroll_id=sid, scroll='5m')
+        sid = search_results['_scroll_id']
+        scroll_size = len(search_results['hits']['hits'])
```
Comment (drjova) [inline scoap3/modules/tools/tools.py:101]: after the while please clear the scroll:

```
current_search_client.clear_scroll(scroll_id=sid)
```

**Pull request merged**